README.md

```diff
@@ -1,2 +1,54 @@
 # golang-ci
-CircleCI orb for performing common tasks from the CI of a project written in Go
+CircleCI orb for performing common CI tasks for a Go project
+
+#### How to use this project
+Please refer to [CircleCI registry](https://circleci.com/orbs/registry/orb/financial-times/golang-ci) for information how to use this orb.
+
+#### How the CI steps for this orb work
+
+There are two CircleCI workflows associated with this project. They have different behaviour depending on whether they are triggered by commit push or by creating/pushing specific git tag.
+1. On commit
+
+When you are pushing a commit in any branch including master, the first workflow is going to: 
+* lint the orb's code
+* packed it into single yaml file 
+* deploy it as "dev:alpha" version. These dev versions of the orb are not visible in the CircleCI registry but can be referred by other projects and can be tested.
+* trigger a new CircleCI pipeline. The new pipeline actually allows the just uploaded dev version of the orb to be used. There is no way to refer to the newly deployed dev version of the orb in the same pipeline that it's been deployed.
+
+When you are pushing a commit, the second workflow is going to:
+* run integration tests for the orb using the just deployed dev version. The integration tests in our cases are actually running the main orb's jobs and ensuring they succeed.
+
+2. On tag
+
+When you are creating a new tag from master branch (you can use the simple GitHub UI to create new release with associated git tag), the first workflow is going to:
+* lint the orb's code
+* packed it into single yaml file 
+* deploy it as "dev:alpha" version
+* trigger a new CircleCI pipeline
+* deploy new version of the orb in the CircleCI registry
+
+In order to trigger the first workflow, your git tag should match the following regular expression `(major|minor|patch)-release-v\d+\.\d+\.\d+`. So if you want your release to upload new patch version of the orb, use tag patch-release-v1.0.1. Unfortunately the numbers in the release are ignored by CircleCI. It only takes into consideration if you tag is `major`, `minor` or `patch` and depending on that it just bumps the current version of the orb in the CircleCI registry. So if your release is `patch-relases-v.1.1.12` and the current version of the orb in the registry is `v1.1.0`, you will actually release `v1.1.1` in the registry.
```

Comment:
If CircleCI cares only about  `major`, `minor`, `patch` why are we requesting from the user to enter a semver as well. Wouldn't it cause more confusion? 

In  `patch-relases-v.1.1.12` it probably should be  without the first dot.

Reply:
We should try to keep our releases in sync with the releases in CircleCI. It is always a good practice have semantic version in the tag name. I was trying to make obvious one of the flows that I found in the CircleCI implementation.
The second one is a typo - I will fix it.

Reply:
I fixed the typo.